pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -332,7 +332,17 @@ func LoadFromFile(filename string) (*clientcmdapi.Config, error) {
 	// set LocationOfOrigin on every Cluster, User, and Context
 	for key, obj := range config.AuthInfos {
 		obj.LocationOfOrigin = filename
+		if len(obj.Token) == 0 && len(obj.TokenFile) > 0 {
```

Comment:
I wasn't expecting this.  I think what you were looking for is down here: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader.go#L532 .  That would let you resolve the relative bindings.